src/components/flowchart-primary-toolbar/flowchart-primary-toolbar.js

```diff
@@ -34,12 +35,17 @@ export const FlowchartPrimaryToolbar = ({
   display,
   visibleLayers,
   expandedPipelines,
+  onToggleExpandPipelines,
   onToggleExpandAllPipelines,
 }) => {
   const { toSetQueryParam } = useGeneratePathname();
 
   const handleToggleExpandAllPipelines = () => {
     const isExpanded = !expandedPipelines;
+
+    // Pass an empty array when collapsing all pipelines
+    onToggleExpandPipelines(isExpanded ? modularPipelineIDs : []);
```

Comment:
over here modularPipelineIDs is an array which also has __root__ so maybe we can explicitly remove root .

Reply:
I've added this but sometimes the flowchart splits and sometimes it doesn't could you check on your end?

Reply:
Also I see that the subpipelines like `Linear Regression` and `Random Forest` still appear.